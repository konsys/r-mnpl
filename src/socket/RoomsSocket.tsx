import { IRoomResponce, setRooms } from "stores/Game/Rooms/RoomsModel";
import React, { useEffect } from "react";

import { SocketActions } from "types/Socket/SocketTypes";
import io from "socket.io-client";
import { SOCKET_PARAMS } from "params/socketParams";

export default function RoomsSocket() {
  useEffect(() => {
    const gameSocket = io(SOCKET_PARAMS.ROOMS_SOCKET);
    gameSocket.on(SocketActions.ROOMS_MESSAGE, (m: IRoomResponce) => {
      try {
        setRooms(m);
      } catch (e) {
        //NOP
      }
    });
    return () => {};
  }, []);
  return <></>;
}
