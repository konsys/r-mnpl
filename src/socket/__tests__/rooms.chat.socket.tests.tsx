import { shallow } from "enzyme";
import RoomsSocket from "socket/RoomsSocket";
import React from "react";

describe("Name of the group", () => {
  it("should render", () => {
    expect(shallow(<RoomsSocket />)).toMatchSnapshot();
  });
});
