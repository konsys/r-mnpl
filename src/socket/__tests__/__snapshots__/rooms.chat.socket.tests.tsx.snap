// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group should render 1`] = `ShallowWrapper {}`;
